### PR TITLE
Ceara/aitt 507 summary rubric carousel

### PR DESCRIPTION
**Pull request by cearachew**
Adds a summary view to the learning goals carousel. The summary shows assessments selected by the teacher for each learning goal. This also fixes a race condition where the progress ring was rendered before all submitted assessments were fetched.

No scores have been selected: 
Screenshot 2024-03-27 at 12 56 40 PM

One score has been selected:
Screenshot 2024-03-27 at 12 57 00 PM

All scores have been selected:
Screenshot 2024-03-27 at 12 56 14 PM

## Links

Jira Ticket: https://codedotorg.atlassian.net/browse/AITT-507

## Testing story

Fixed and added unit tests for this feature

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/rubrics/LearningGoals.jsx [hunk 1]
```diff
@@ -442,18 +458,21 @@ export default function LearningGoals({
...
 
-    if (!isStudent) {
-      const eventName = EVENTS.TA_RUBRIC_LEARNING_GOAL_SELECTED;
-      analyticsReporter.sendEvent(eventName, {
-        ...(reportingData || {}),
-        learningGoalKey: learningGoals[currentIndex].key,
-        learningGoal: learningGoals[currentIndex].learningGoal,
-      });
+      if (!isStudent) {
+        console.group(`Send Event`);
```
Comment (Nokondi) [inline apps/src/templates/rubrics/LearningGoals.jsx]: Do you need to ungroup at the end of this event?

**Review by Nokondi — COMMENTED** [review]
Just one question about the console.group() usage, but otherwise looks great!

File: apps/src/templates/rubrics/LearningGoals.jsx (hunk 1 above)
Comment (cearachew) [reply apps/src/templates/rubrics/LearningGoals.jsx]: oops I think this is left over from debugging, lemme remove it

File: apps/src/templates/rubrics/LearningGoals.jsx [hunk 2]
```diff
@@ -352,6 +363,9 @@ export default function LearningGoals({
                 setLoaded(teacherFeedbacksLoaded.current[index]);
                 setDisplayUnderstanding(understandingLevels.current[index]);
               }
+              if (index === learningGoals.length - 1) {
+                setDoneLoading(true);
+              }
```
Comment (Nokondi) [inline apps/src/templates/rubrics/LearningGoals.jsx:368]: I like this solution to the race condition you were experiencing. I actually had to use something similar for the product tour to make sure the rubric is fully loaded before the tour starts.

**Review by Nokondi — APPROVED** [review]
Thanks for taking care of this!

**Pull request merged**